GeoSpatial Configuration Updates

Michael7371 opened the pull request:
Updates to support geospatially updating RSU SNMP forwarding.

For testing feel free to use the updated API that is already deployed in cloud run by adding the sub-string "config---" between "https://" and "rsu-manager" in the REACT_APP_GATEWAY_BASE_URL_DEV environmental variable.

Michael7371 committed:
Delete SNMP Command

Added delete snmp functionality within the configuration of RSU on main page

Michael7371 committed:
api updates for geospatial query of RSUs

Michael7371 committed:
updates to api for config stuff

debbieneaeraconsulting committed:
Merge remote-tracking branch 'origin/configuration-geo-query-updates' into configration_changes_deletesnmp

debbieneaeraconsulting committed:
updates to fix polygon rendering

Michael7371 committed:
updates from api deployment

Michael7371 committed:
Merge remote-tracking branch 'origin/configration_changes_deletesnmp' into configuration-updates

Michael7371 committed:
ui updates for multi rsu config

Michael7371 committed:
updates to api, moved geospatial rsu query stuff from rsu slice to config slice

Michael7371 committed:
fixes to snmpset menu

Michael7371 committed:
Merge branch 'develop' into configuration-updates

debbieneaeraconsulting committed:
updated config box height

updated config box height

Michael7371 committed:
updates to the geospatial configuration page.

Michael7371 committed:
updates to the configuration menu

debbieneaeraconsulting committed:
updates to map page to only allow one toggle

debbieneaeraconsulting committed:
separate point/selection for bsms and rsu configurations

Michael7371 committed:
api unit tests start, ui fixes

Michael7371 committed:
fixed bsm data issues, added some unit tests, and updated the menu code

Michael7371 committed:
unit tests

debbieneaeraconsulting committed:
Fixed disabled point selection

debbieneaeraconsulting committed:
Merge branch 'configuration-updates' of https://github.com/CDOT-CV/jpo-cvmanager into configuration-updates

Michael7371 committed:
updated css for map button

Michael7371 committed:
updates to bsm viewer

debbieneaeraconsulting committed:
Merge branch 'configuration-updates' of https://github.com/CDOT-CV/jpo-cvmanager into configuration-updates

drewjj commented:
![image](https://user-images.githubusercontent.com/31270488/237016249-9449e0d7-177e-4c0a-ab4e-8cdedaf5900c.png)
Maybe we would want to automatically close the multi-select if we select another single RSU.

drewjj commented:
Why did we add RSU indexes back into the SNMP configuration menus? This was originally abstracted to be automatically handled by the API since this concept is not something we want the user of the CV Manager to need to understand.

drewjj commented:
![image](https://user-images.githubusercontent.com/31270488/237017269-040228ea-4b0a-4778-9df6-e0a400a0609c.png)
I don't mind this menu but I feel like the UX choices are a little convoluted. I wonder if this can be streamlined to be more intuitive. Maybe hide the X until there is something to clear out on the map and hide it again when it is cleared. 

If you turn the toggle on, add some points to create a complete polygon, and then turn it off before clicking the "CONFIGURE RSU'S" you will get it in some sort of soft lock where it no longer works until you clear it. Maybe finding a way to consolidate the menu here into less independent buttons and more of a stateful menu which helps guide the user through the correct manner of using the UX would be nice.

Also, I feel like we should change the text to "Configure RSUs" without the apostrophe and only capitalizing the first letter of each word.  All of the other buttons use lowercase letters so this button is pretty unique which kinda contradicts our CSS choices at the moment. We could always redesign our CSS to be different but I think that can be saved for a different pull request.

drewjj reviewed (changes requested):
My requests are reflected in my comments. They are more UX related so I made them comments to add pictures. I am happy to help provide some ideas with the UX stuff if you want to reach out to me!

Michael7371 commented:
> ![image](https://user-images.githubusercontent.com/31270488/237016249-9449e0d7-177e-4c0a-ab4e-8cdedaf5900c.png)
> Maybe we would want to automatically close the multi-select if we select another single RSU.

Yeah I like that idea.

Michael7371 commented:
> Why did we add RSU indexes back into the SNMP configuration menus? This was originally abstracted to be automatically handled by the API since this concept is not something we want the user of the CV Manager to need to understand.

I thought it was needed for the deletion endpoint, if it isn't then we can get rid of it.

drewjj commented:
> > Why did we add RSU indexes back into the SNMP configuration menus? This was originally abstracted to be automatically handled by the API since this concept is not something we want the user of the CV Manager to need to understand.
> 
> I thought it was needed for the deletion endpoint, if it isn't then we can get rid of it.

I do not believe it is required. The API's current deletion logic should loop through the RSUs, perform basic snmpwalks on all of them and identify the index that matches the specified message type and destination IP and then delete that index.

Michael7371 commented:
> > > Why did we add RSU indexes back into the SNMP configuration menus? This was originally abstracted to be automatically handled by the API since this concept is not something we want the user of the CV Manager to need to understand.
> > 
> > 
> > I thought it was needed for the deletion endpoint, if it isn't then we can get rid of it.
> 
> I do not believe it is required. The API's current deletion logic should loop through the RSUs, perform basic snmpwalks on all of them and identify the index that matches the specified message type and destination IP and then delete that index.

Awesome! I will get that updated

Michael7371 committed:
addressing PR comments

Michael7371 commented:
@drewjj I just pushed a commit that addresses most of your comments. Also, you were right about the index not being required, I didn't see that the rsu_commands.py has the functionality of getting the message index. I was able to hide the clear button when it isn't useful, as well as changed the text format to not be in all caps. The configure menu was also updated to hide the multi-rsu selection if a single rsu is selected. I also added a clear button for the single rsu config menu to keep the menu view consistent.

Michael7371 committed:
initial fixes for snmp deletion

Michael7371 committed:
updates with fixed deletion and other ui fixes

Michael7371 committed:
pr updates for multi delete of snmp forwarded endpoints.

drewjj reviewed (approved):
Awesome, it behaves nicely and works well. Great job!

Michael7371 committed:
Merge branch 'develop' into configuration-updates

Michael7371 committed:
fixing merge conflict. there are still problems with rsu info popup

Michael7371 committed:
fixing popups for rsus. No longer closes when clicking the map though.

Michael7371 committed:
updates to css and unit tests

Michael7371 committed:
updating variables in configTest

drewjj committed:
Implement workaround for MapboxGL bug with dependencies on latest version

jacob6838 committed:
Allowing rsu and wzdx popups to close on map click

drewjj reviewed (approved):
Re-approving to show I have tested the current state of the pull request since my old approval is stale.

payneBrandon commented on api/.gitignore:
```diff
@@ -0,0 +1,2 @@
+env
```
Should this be ".env"?

Michael7371 replied:
I added it because I was using a virtual environment. I'll add .env as well.

Michael7371 committed:
Merge branch 'develop' into configuration-updates

Michael7371 committed:
updates for unit tests and code coverage

Michael7371 committed:
added description to multi rsu add/remove snmp configuration box

Michael7371 committed:
updated unit tests

payneBrandon commented on webapp/src/components/SnmpwalkMenu.js:
```diff
@@ -26,44 +35,164 @@ const SnmpwalkMenu = (props) => {
+                {Object.keys(msgFwdConfig.rsuReceivedMsgTable).map((index) => (
+                  <div>
+                    <Button
+                      className="deletbutton"
```
Is this the correct class name?

payneBrandon replied:
Looks like it reference the right css class, just misspelled in both locations. We can address this later, closing.

payneBrandon reviewed (approved):
Looking good! A couple clarifying questions is all
